.gitignore

```diff
@@ -16,6 +16,11 @@ testMain
 # CLion
 /.idea
 .vscode/
+# Created in vs2017 when using open folder
+.vs/
+
+# Created in vs2017 when using open folder with cmake tools
+CMakeSettings.json
```

Comment:
I think this only happens for in source builds, which we do not recommand
https://github.com/xbmc/xbmc/blob/3f32f55ee7139fe3b4132b292986d87f925d71c4/CMakeLists.txt#L19-L21

Reply:
It's created by VS to store settings on how it should configure cmake. It's a bit weird that they don't stick it in the .vs folder.
Here's a sample of of
```
{
    // See https://go.microsoft.com//fwlink//?linkid=834763 for more information about this file.
    "configurations": [
        {
            "name": "x64-Debug",
            "generator": "Ninja",
            "configurationType": "Debug",
            "inheritEnvironments": [ "msvc_x64_x64" ],
            "buildRoot": "${env.USERPROFILE}\\CMakeBuilds\\${workspaceHash}\\build\\${name}",
            "installRoot": "${env.USERPROFILE}\\CMakeBuilds\\${workspaceHash}\\install\\${name}",
            "cmakeCommandArgs": "",
            "buildCommandArgs": "-v",
            "ctestCommandArgs": ""
        },
        {
            "name": "x64-Release",
            "generator": "Ninja",
            "configurationType": "RelWithDebInfo",
            "inheritEnvironments": [ "msvc_x64_x64" ],
            "buildRoot": "${env.USERPROFILE}\\CMakeBuilds\\${workspaceHash}\\build\\${name}",
            "installRoot": "${env.USERPROFILE}\\CMakeBuilds\\${workspaceHash}\\install\\${name}",
            "cmakeCommandArgs": "",
            "buildCommandArgs": "-v",
            "ctestCommandArgs": ""
        },
        {
            "name": "Linux-Debug",
            "generator": "Unix Makefiles",
            "remoteMachineName": "${defaultRemoteMachineName}",
            "configurationType": "Debug",
            "remoteCMakeListsRoot": "/var/tmp/src/${workspaceHash}/${name}",
            "cmakeExecutable": "/usr/local/bin/cmake",
            "buildRoot": "${env.USERPROFILE}\\CMakeBuilds\\${workspaceHash}\\build\\${name}",
            "installRoot": "${env.USERPROFILE}\\CMakeBuilds\\${workspaceHash}\\install\\${name}",
            "remoteBuildRoot": "/var/tmp/build/${workspaceHash}/build/${name}",
            "remoteInstallRoot": "/var/tmp/build/${workspaceHash}/install/${name}",
            "remoteCopySources": true,
            "remoteCopySourcesOutputVerbosity": "Normal",
            "remoteCopySourcesConcurrentCopies": "10",
            "remoteCopySourcesMethod": "rsync",
            "cmakeCommandArgs": "",
            "buildCommandArgs": "",
            "ctestCommandArgs": "",
            "inheritEnvironments": [ "linux-x64" ]
        },
        {
            "name": "Linux-Release",
            "generator": "Unix Makefiles",
            "remoteMachineName": "${defaultRemoteMachineName}",
            "configurationType": "Release",
            "remoteCMakeListsRoot": "/var/tmp/src/${workspaceHash}/${name}",
            "cmakeExecutable": "/usr/local/bin/cmake",
            "buildRoot": "${env.USERPROFILE}\\CMakeBuilds\\${workspaceHash}\\build\\${name}",
            "installRoot": "${env.USERPROFILE}\\CMakeBuilds\\${workspaceHash}\\install\\${name}",
            "remoteBuildRoot": "/var/tmp/build/${workspaceHash}/build/${name}",
            "remoteInstallRoot": "/var/tmp/build/${workspaceHash}/install/${name}",
            "remoteCopySources": true,
            "remoteCopySourcesOutputVerbosity": "Normal",
            "remoteCopySourcesConcurrentCopies": "10",
            "remoteCopySourcesMethod": "rsync",
            "cmakeCommandArgs": "",
            "buildCommandArgs": "",
            "ctestCommandArgs": "",
            "inheritEnvironments": [ "linux-x64" ]
        }
    ]
}
```